crates/zk-por-core/src/merkle_sum_tree.rs

```diff
@@ -42,15 +43,21 @@ pub struct MerkleSumTree {
 }
 
 impl MerkleSumTree {
-    /// Create a new merkle sum tree from a set of accounts, note that this set must be a power of 2.
-    /// In the future we can try to pad with empty accounts.
-    pub fn new_tree_from_accounts(accounts: &Vec<Account>) -> MerkleSumTree {
+    pub fn new_tree_from_accounts(accounts: &Vec<Account>, num_leaves: usize) -> MerkleSumTree {
```

Comment:
I would suggest to make out the padding logic to the caller of this function. Otherwise, padding inside this function could surprise the caller. Instead, this function merely asserts the number of accounts is in power of 2.